lib/src/core/models/chat/sub_models/choices/sub_models/message.dart

```diff
@@ -19,6 +23,7 @@ final class OpenAIChatCompletionChoiceMessageModel {
   const OpenAIChatCompletionChoiceMessageModel({
     required this.role,
     required this.content,
+    this.functionCall,
```

Comment:
what do you think about having a getter to return a `bool` if the `functionCall` exists or not such as:
```dart
bool get requiresFunctionCall => functionCall != null;
```